debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/JdbcOffsetBackingStore.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.storage.jdbc;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.common.utils.ThreadUtils;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.OffsetBackingStore;
+import org.apache.kafka.connect.util.Callback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Field;
+
+/**
+ * Implementation of OffsetBackingStore that saves data to database table.
+ */
+public class JdbcOffsetBackingStore implements OffsetBackingStore {
+
+    public static final Field JDBC_URI = Field.create("offset.storage.jdbc.uri")
+            .withDescription("Uri of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    public static final Field JDBC_USER = Field.create("offset.storage.jdbc.user")
+            .withDescription("Username of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    public static final Field JDBC_PASSWORD = Field.create("offset.storage.jdbc.password")
+            .withDescription("Password of the database which will be used to record the database history")
+            .withValidation(Field::isRequired);
+
+    public static final String OFFSET_STORAGE_TABLE_NAME = "debezium_offset_storage";
```

Comment:
Please make both table name and create statement configurable with this default. The reason is that the user might have presirbed naming policy and we cannot be sure that this statement will work on an arbitrary database so it is better to provide a way how to override it.
Tha same is for database history.

Reply:
@ismailsimsek Any opinion on this comment?

Reply:
@jpechane I understand the part of making the OFFSET_STORAGE_TABLE_NAME configurable, but if we make the CREATE TABLE sql configurable then its hard to read the column names.
Can we just make the column names configurable with defaults.

Like `offset.storage.jdbc.offset_key_column `, `offset.storage.jdbc.offset_val_column`,  thoughts?